paimon-core/src/main/java/org/apache/paimon/table/PrimaryKeyFileStoreTable.java

```diff
@@ -185,7 +186,8 @@ protected List<CommitCallback> createCommitCallbacks(String commitUser) {
         List<CommitCallback> callbacks = super.createCommitCallbacks(commitUser);
         CoreOptions options = coreOptions();
 
-        if (options.metadataIcebergCompatible()) {
+        if (options.toConfiguration().get(IcebergOptions.METADATA_ICEBERG_STORAGE)
+                != IcebergOptions.StorageType.DISABLED) {
```

Comment:
Move this to AbstractFileStoreTable for AppendOnlyFileStoreTable and PrimaryKeyFileStoreTable?